### ISSUE TITLE
Use getItem instead of iterate for small getItem array lookups

### ISSUE THREAD

**Issue by scotttrinh** [ocombe/angular-localForage#134]
Currently, even if you're only getting a single kv pair, if that lookup is in an array, it will iterate the whole database.

We should look into just returning a `$q.all` with the lookup of the exactly keys instead of skipping, or if the `keys` lookup is cheap, find some heuristic to know when it's cheaper to use `iterate` vs. multiple `getItem` calls.

**Comment by pablorsk** [comment]
I found [LocalForage-getItems](https://github.com/localForage/localForage-getItems), but I dont know how use it with angular-localForage.

**Comment by scotttrinh** [comment]
Yeah, you'd have to set it up in a `config` block or something like that. I think there is a way to accomplish a similar thing as that extension, but I think we want to stick with vanilla localForage so we don't have to burden users with another dependency. Thanks for pointing that out, though! I'm sure there will be some useful things to learn from reading through that code.
